tests/tools/tools/serialize.fs

```diff
@@ -291,4 +297,95 @@ module SerializationTests =
     let testNonSerializable() = 
         testObject <| MyType()
         
+    let timeRoundTrip (name: string) (formatter: IFormatter) (stream: MemoryStream) obj =
+        GC.Collect()
+        let sw = Stopwatch.StartNew()
+        do formatter.Serialize(stream, obj)
+        let serTime = sw.Elapsed
+        stream.Position <- 0L
+        sw.Restart()
+        let ret = formatter.Deserialize(stream) 
+        let deserTime = sw.Elapsed
+        serTime, deserTime
+
+    let memStreamConstructors = memStreamBConstructors
+    let noArgConstructor = fst memStreamBConstructors
+
+    let compareRoundTripTimes : int -> obj -> TimeSpan * TimeSpan = 
+        let selector = CustomizedSerializationSurrogateSelector(memStreamConstructors)
+        let prajnaFormatter = GenericSerialization.GetFormatter(GenericSerialization.PrajnaFormatterGuid, selector)
+        let binaryFormatter = GenericSerialization.GetFormatter(GenericSerialization.BinaryFormatterGuid, selector)
+        fun numRepeats obj ->
+            let accPair (f,s) (f2,s2) = f + f2, s + s2
+            let sumTimePairs = Seq.fold accPair (TimeSpan.Zero, TimeSpan.Zero)
+            let binarySer,binaryDeser = [for i in 1..numRepeats -> timeRoundTrip "Binary" binaryFormatter (noArgConstructor()) obj] |> sumTimePairs
+            let prajnaSer,prajnaDeser = [for i in 1..numRepeats -> timeRoundTrip "Prajna" prajnaFormatter (noArgConstructor()) obj] |> sumTimePairs
+            Console.WriteLine(sprintf "Prajna: %A, %A, %A" prajnaSer prajnaDeser (prajnaSer + prajnaDeser))
+            Console.WriteLine(sprintf "Binary: %A, %A, %A" binarySer binaryDeser (binarySer + binaryDeser))
+            (prajnaSer + prajnaDeser), (binarySer + binaryDeser)
+    
+    [<Test>]
+    [<Category("Performance")>]
+    let SerPerfMatrix() =
+        printfn "Square Matrix"
+        let r = new Random()
+        let A : float[,] = Array2D.init 5000 5000 (fun _ _ -> r.NextDouble() |> float)
+        let prajnaTime, binaryTime = compareRoundTripTimes 1 A
+        Assert.IsTrue(prajnaTime.TotalMilliseconds < binaryTime.TotalMilliseconds)
+        
+    [<Test>]
+    [<Category("Performance")>]
+    let SerPerfRank3Tensor() =
+        printfn "Cube Tensor"
+        let r = new Random()
+        let A : float[,,] = Array3D.init 300 300 300 (fun _ _ _ -> r.NextDouble() |> float)
+        let prajnaTime, binaryTime = compareRoundTripTimes 1 A
+        Assert.IsTrue(prajnaTime.TotalMilliseconds < binaryTime.TotalMilliseconds)
+        
+    let createRandomJaggedMatrix rows cols = 
+        let r = new Random()
+        Array.init rows (fun _ -> Array.init cols (fun _ -> r.NextDouble() |> float))
+
+    let baseSize = 5000
     
+    [<Test>]
+    [<Category("Performance")>]
```

Comment:
These tests will not be ran as part of the build since they are excluded. But they will show up in VS test explorer, which probably does not have the functionality to exclude based on NUnit category.  If that's the case, we need to add some note to the wiki (https://github.com/MSRCCS/Prajna/wiki/Contribute, at the "Visual Studio" section) to call it out.  So when ppl do a "run all" in test explorer, they won't be surprised.